### ISSUE TITLE
Segmentation fault with ffmpeg on Debian 12 and derivatives 

### ISSUE THREAD

**Issue by glyphscribe** [owncast/owncast#4045]
### summary
_Owncast_ installations across a variety of Debian 12 hosts fail to generate thumbnails for an active stream due to an underlying segfault of _ffmpeg_. libx264 is in use, and normal stream video transcoding is not apparently affected. Ubuntu 24 hosts may also be affected, but I have not verified that.

### environment information:
This issue has been tested across a variety of _virtualized_ Debian 12 installations using different hypervisors and virtualized hardware, with the issue persisting identically between all tested. In all cases, owncast was installed using the quickstart script provided on the owncast website. The originating stream in testing is being provided by OBS with `nvenc h.264` encoding, but the issue appears to persist across varied encoding strategies.

Preparing a Debian 11 installation in the same environment(s) with the same virtualized hardware and an identical OS and Owncast installation methodology yields no issues -- the full suite works as-expected with the bundled ffmpeg and functional thumbnail generation.

### log information
Owncast stream log (repeats every 20 seconds during stream):
```
Unable to generate thumbnail: Unable to generate thumbnail: exit status 139
```

Transcoder log:
```
ffmpeg started on 2024-12-04 at 17:00:34
Report written to "data/logs/transcoder.log"
Log level: 32
Command line:
/usr/local/bin/ffmpeg -hide_banner -loglevel warning -fflags +genpts -flags +cgo       p -i pipe:0 -map v:0 -c:v:0 libx264 -b:v:0 2808k -maxrate:v:0 3032k -g:v:0 90 -k       eyint_min:v:0 90 -r:v:0 30 -x264-params:v:0 "scenecut=0:open_gop=0" -bufsize:v:0        3032k -profile:v:0 high -map "a:0?" -c:a:0 copy -preset fast -var_stream_map "v       :0,a:0 " -f hls -hls_time 3 -hls_list_size 10 -hls_flags program_date_time+indep       endent_segments+omit_endlist -segment_format_options "mpegts_flags=mpegts_copyts       =1" -tune zerolatency -pix_fmt yuv420p -sc_threshold 0 -master_pl_name stream.m3       u8 -hls_segment_filename "http://127.0.0.1:38245/%v/stream-vryxjcVHRm-%d.ts" -ma       x_muxing_queue_size 400 -method PUT "http://127.0.0.1:38245/%v/stream.m3u8"
Input #0, flv, from 'pipe:0':
  Metadata:
    fileSize        : 0
    audiochannels   : 2
    2.1             : false
    3.1             : false
    4.0             : false
    4.1             : false
    5.1             : false
    7.1             : false
    encoder         : obs-output module (libobs version 30.2.3)
  Duration: 00:00:00.00, start: 0.066000, bitrate: N/A
  Stream #0:0: Audio: aac (LC), 48000 Hz, stereo, fltp, 327 kb/s
  Stream #0:1: Video: h264 (High), yuv420p(tv, bt709, progressive), 1664x936 [SA       R 1:1 DAR 16:9], 6144 kb/s, 30 fps, 30 tbr, 1k tbn
Stream mapping:
  Stream #0:1 -> #0:0 (h264 (native) -> h264 (libx264))
  Stream #0:0 -> #0:1 (copy)
[libx264 @ 0x2c3456c0] using SAR=1/1
[libx264 @ 0x2c3456c0] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2
[libx264 @ 0x2c3456c0] profile High, level 4.0, 4:2:0, 8-bit
[libx264 @ 0x2c3456c0] 264 - core 164 r3161 a354f11 - H.264/MPEG-4 AVC codec - C       opyleft 2003-2023 - http://www.videolan.org/x264.html - options: cabac=1 ref=2 d       eblock=1:0:0 analyse=0x3:0x113 me=hex subme=6 psy=1 psy_rd=1.00:0.00 mixed_ref=1        me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 ch       roma_qp_offset=-2 threads=6 lookahead_threads=6 sliced_threads=1 slices=6 nr=0 d       ecimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=0 weightp=1 k       eyint=90 keyint_min=46 scenecut=0 intra_refresh=0 rc_lookahead=0 rc=abr mbtree=0        bitrate=2808 ratetol=1.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 vbv_maxrate=3032        vbv_bufsize=3032 nal_hrd=none filler=0 ip_ratio=1.40 aq=1:1.00
Output #0, hls, to 'http://127.0.0.1:38245/%v/stream.m3u8':
  Metadata:
    fileSize        : 0
    audiochannels   : 2
    2.1             : false
    3.1             : false
    4.0             : false
    4.1             : false
    5.1             : false
    7.1             : false
    encoder         : Lavf60.16.100
  Stream #0:0: Video: h264, yuv420p(tv, bt709, progressive), 1664x936 [SAR 1:1 D       AR 16:9], q=2-31, 2808 kb/s, 30 fps, 90k tbn
    Metadata:
      encoder         : Lavc60.31.102 libx264
    Side data:
      cpb: bitrate max/min/avg: 3032000/0/2808000 buffer size: 3032000 vbv_delay       : N/A
  Stream #0:1: Audio: aac (LC), 48000 Hz, stereo, fltp, 327 kb/s
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-0.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-1.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-2.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-3.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-4.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-5.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-6.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-7.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-8.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-9.ts' for        writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-10.ts' fo       r writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-11.ts' fo       r writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-12.ts' fo       r writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream-vryxjcVHRm-13.ts' fo       r writing
[tcp @ 0x2c44da00] Connection to tcp://127.0.0.1:38245 failed: Connection refuse       d
[hls @ 0x2c342f40] Failed to open file 'http://127.0.0.1:38245/0/stream-vryxjcVH       Rm-13.ts'
[hls @ 0x2c342f40] Opening 'http://127.0.0.1:38245/0/stream.m3u8' for writing
[tcp @ 0x2c44da00] Connection to tcp://127.0.0.1:38245 failed: Connection refuse       d
[out#0/hls @ 0x2c560080] video:14019kB audio:1280kB subtitle:0kB other streams:0       kB global headers:0kB muxing overhead: unknown
frame= 1221 fps= 31 q=29.0 Lsize=N/A time=00:00:40.85 bitrate=N/A speed=1.03x          
[libx264 @ 0x2c3456c0] frame I:14    Avg QP:21.77  size: 46685
[libx264 @ 0x2c3456c0] frame P:1207  Avg QP:23.75  size: 11352
[libx264 @ 0x2c3456c0] mb I  I16..4: 11.6% 83.7%  4.7%
[libx264 @ 0x2c3456c0] mb P  I16..4:  3.9% 10.7%  0.0%  P16..4: 39.2%  3.2%  2.7       %  0.0%  0.0%    skip:40.4%
[libx264 @ 0x2c3456c0] 8x8 transform intra:74.1% inter:95.8%
[libx264 @ 0x2c3456c0] coded y,uvDC,uvAC intra: 36.7% 56.1% 0.8% inter: 12.2% 25       .8% 0.0%
[libx264 @ 0x2c3456c0] i16 v,h,dc,p:  6% 43% 16% 35%
[libx264 @ 0x2c3456c0] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu:  8% 16% 56%  2%  4%  3%  5       %  2%  5%
[libx264 @ 0x2c3456c0] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 17% 35% 24%  4%  5%  3%  5       %  2%  5%
[libx264 @ 0x2c3456c0] i8c dc,h,v,p: 61% 25% 12%  2%
[libx264 @ 0x2c3456c0] Weighted P-Frames: Y:0.2% UV:0.0%
[libx264 @ 0x2c3456c0] ref P L0: 55.2% 44.8%
[libx264 @ 0x2c3456c0] kb/s:2821.66
Exiting normally, received signal 2.
```

Relevant lines in dmesg (every 20s):
```
[839678.090618] ffmpeg[95963]: segfault at 270e0 ip 00000000000270e0 sp 00007fff7b5c5558 error 14 in ffmpeg[400000+1000] likely on CPU 4 (core 4, socket 0)
[839678.090629] Code: Unable to access opcode bytes at 0x270b6.
```

### ffmpeg version research
The issue persists whether or not the bundled version of ffmpeg is used. Version dumps below:

bundled 4.3.1
```
ffmpeg version 4.3.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2020 the FFmpeg developers
built with gcc 8 (Debian 8.3.0-6)
configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
libavutil      56. 51.100 / 56. 51.100
libavcodec     58. 91.100 / 58. 91.100
libavformat    58. 45.100 / 58. 45.100
libavdevice    58. 10.100 / 58. 10.100
libavfilter     7. 85.100 /  7. 85.100
libswscale      5.  7.100 /  5.  7.100
libswresample   3.  7.100 /  3.  7.100
libpostproc    55.  7.100 / 55.  7.100
```

system 6.1
```
ffmpeg version 6.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2023 the FFmpeg developers
built with gcc 8 (Debian 8.3.0-6)
configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
libavutil      58. 29.100 / 58. 29.100
libavcodec     60. 31.102 / 60. 31.102
libavformat    60. 16.100 / 60. 16.100
libavdevice    60.  3.100 / 60.  3.100
libavfilter     9. 12.100 /  9. 12.100
libswscale      7.  5.100 /  7.  5.100
libswresample   4. 12.100 /  4. 12.100
libpostproc    57.  3.100 / 57.  3.100
```

### other information

Tagging @flexiondotorg per their request, and @drfrankenspine because much of the original testing was his data. I can perform bare metal hardware testing if necessary going forward.

**Comment by glyphscribe** [comment]
### Supplemental information
Below are traces of `ffmpeg` invocations simulating owncast's internal calls to generate thumbnails:

**Debian 11, success**:
```
user@owncast-test-11:~/owncast/data/tmp$ ../../ffmpeg -y -threads 1 -t 1 -i offline.ts583003426 -f image2 -vframes 1 test.png -loglevel trace
ffmpeg version 4.3.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Splitting the commandline.
Reading option '-y' ... matched as option 'y' (overwrite output files) with argument '1'.
Reading option '-threads' ... matched as AVOption 'threads' with argument '1'.
Reading option '-t' ... matched as option 't' (record or transcode "duration" seconds of audio/video) with argument '1'.
Reading option '-i' ... matched as input url with argument 'offline.ts583003426'.
Reading option '-f' ... matched as option 'f' (force format) with argument 'image2'.
Reading option '-vframes' ... matched as option 'vframes' (set the number of video frames to output) with argument '1'.
Reading option 'test.png' ... matched as output url.
Reading option '-loglevel' ... matched as option 'loglevel' (set logging level) with argument 'trace'.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option y (overwrite output files) with argument 1.
Applying option loglevel (set logging level) with argument trace.
Successfully parsed a group of options.
Parsing a group of options: input url offline.ts583003426.
Applying option t (record or transcode "duration" seconds of audio/video) with argument 1.
Successfully parsed a group of options.
Opening an input file: offline.ts583003426.
[NULL @ 0xfb9e440] Opening 'offline.ts583003426' for reading
[file @ 0xfb9ef00] Setting default whitelist 'file,crypto,data'
Probing mpegts score:50 size:2048
[mpegts @ 0xfb9e440] Format mpegts probed with size=2048 and score=50
[mpegts @ 0xfb9e440] Probe: 8192, score: 44, dvhs_score: 0, fec_score: -2
[mpegts @ 0xfb9e440] Filter: pid=0x11 type=1
[mpegts @ 0xfb9e440] Filter: pid=0x0 type=1
[mpegts @ 0xfb9e440] Filter: pid=0x12 type=1
[mpegts @ 0xfb9e440] SDT:
[mpegts @ 0xfb9e440] tag: 0x48 len=24
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] PAT:
[mpegts @ 0xfb9e440] sid=0x1 pid=0x1000
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] Filter: pid=0x1000 type=1
[mpegts @ 0xfb9e440] PMT: len 32
[mpegts @ 0xfb9e440] sid=0x1 sec_num=0/0 version=0 tid=2
[mpegts @ 0xfb9e440] pcr_pid=0x100
[mpegts @ 0xfb9e440] Filter: pid=0x100 type=0
[mpegts @ 0xfb9e440] stream=0 stream_type=1b pid=100 prog_reg_desc=
[mpegts @ 0xfb9e440] Filter: pid=0x101 type=0
[mpegts @ 0xfb9e440] stream=1 stream_type=f pid=101 prog_reg_desc=
[mpegts @ 0xfb9e440] tag: 0x0a len=4
[mpegts @ 0xfb9e440] tuning done
[mpegts @ 0xfb9e440] Before avformat_find_stream_info() pos: 0 bytes read:32768 seeks:0 nb_streams:2
[mpegts @ 0xfb9e440] Skipping after seek
[mpegts @ 0xfb9e440] SDT:
[mpegts @ 0xfb9e440] tag: 0x48 len=24
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] PAT:
[mpegts @ 0xfb9e440] sid=0x1 pid=0x1000
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] PMT: len 32
[mpegts @ 0xfb9e440] sid=0x1 sec_num=0/0 version=0 tid=2
[mpegts @ 0xfb9e440] pcr_pid=0x100
[mpegts @ 0xfb9e440] tag: 0x0a len=4
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[extract_extradata @ 0xfba6100] nal_unit_type: 9(AUD), nal_ref_idc: 0
[extract_extradata @ 0xfba6100] nal_unit_type: 7(SPS), nal_ref_idc: 3
[extract_extradata @ 0xfba6100] nal_unit_type: 8(PPS), nal_ref_idc: 3
[extract_extradata @ 0xfba6100] nal_unit_type: 6(SEI), nal_ref_idc: 0
[extract_extradata @ 0xfba6100] nal_unit_type: 5(IDR), nal_ref_idc: 3
[h264 @ 0xfba5cc0] nal_unit_type: 9(AUD), nal_ref_idc: 0
[h264 @ 0xfba5cc0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0xfba5cc0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0xfba5cc0] nal_unit_type: 6(SEI), nal_ref_idc: 0
[h264 @ 0xfba5cc0] nal_unit_type: 5(IDR), nal_ref_idc: 3
[h264 @ 0xfba5cc0] Format yuv420p chosen by get_format().
[h264 @ 0xfba5cc0] Reinit context to 864x480, pix_fmt: yuv420p
[h264 @ 0xfba5cc0] no picture
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 6 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] probing stream 1 pp:2500
Probing aac score:51 size:221
[mpegts @ 0xfb9e440] Probe with size=221, packets=1 detected aac with score=51
[mpegts @ 0xfb9e440] probed stream 1
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 8 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 8 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
[mpegts @ 0xfb9e440] max_analyze_duration 5000000 reached at 5000000 microseconds st:0
[mpegts @ 0xfb9e440] Skipping after seek
[mpegts @ 0xfb9e440] Probe: 8192, score: 44, dvhs_score: -3, fec_score: -3
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 2 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 4 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 3 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 4 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 7 times
[mpegts @ 0xfb9e440] pid=101 pes_code=0x1c0
    Last message repeated 1 times
[mpegts @ 0xfb9e440] stream 0: start_time: 1.48333 duration: 8.04167
[mpegts @ 0xfb9e440] stream 1: start_time: 1.43933 duration: 8.04267
[mpegts @ 0xfb9e440] format: start_time: 1.43933 duration: 8.08567 (estimate from pts) bitrate=1135 kb/s
[mpegts @ 0xfb9e440] After avformat_find_stream_info() pos: 0 bytes read:1298576 seeks:2 frames:362
Input #0, mpegts, from 'offline.ts583003426':
  Duration: 00:00:08.09, start: 1.439333, bitrate: 1135 kb/s
  Program 1
    Metadata:
      service_name    : Owncast Offline
      service_provider: FFmpeg
    Stream #0:0[0x100], 122, 1/90000: Video: h264 (High), 1 reference frame ([27][0][0][0] / 0x001B), yuv420p(progressive, left), 854x480 (864x480) [SAR 1:1 DAR 427:240], 0/1, 24 fps, 24 tbr, 90k tbn, 48 tbc
    Stream #0:1[0x101](eng), 240, 1/90000: Audio: aac (LC) ([15][0][0][0] / 0x000F), 48000 Hz, stereo, fltp, 155 kb/s
Successfully opened the file.
Parsing a group of options: output url test.png.
Applying option f (force format) with argument image2.
Applying option vframes (set the number of video frames to output) with argument 1.
Successfully parsed a group of options.
Opening an output file: test.png.
Successfully opened the file.
[h264 @ 0xfcf23c0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0xfcf23c0] nal_unit_type: 8(PPS), nal_ref_idc: 3
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> png (native))
Press [q] to stop, [?] for help
cur_dts is invalid st:0 (0) [init:0 i_done:0 finish:0] (this is harmless if it occurs once at the start per stream)
[mpegts @ 0xfb9e440] Skipping after seek
[mpegts @ 0xfb9e440] SDT:
[mpegts @ 0xfb9e440] tag: 0x48 len=24
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] PAT:
[mpegts @ 0xfb9e440] sid=0x1 pid=0x1000
[mpegts @ 0xfb9e440] new_program: id=0x0001
[mpegts @ 0xfb9e440] PMT: len 32
[mpegts @ 0xfb9e440] sid=0x1 sec_num=0/0 version=0 tid=2
[mpegts @ 0xfb9e440] pcr_pid=0x100
[mpegts @ 0xfb9e440] tag: 0x0a len=4
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
    Last message repeated 1 times
[NULL @ 0xfba5cc0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[NULL @ 0xfba5cc0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
[mpegts @ 0xfb9e440] Correcting start time by 44000
[h264 @ 0xfcf23c0] nal_unit_type: 9(AUD), nal_ref_idc: 0
[h264 @ 0xfcf23c0] nal_unit_type: 7(SPS), nal_ref_idc: 3
[h264 @ 0xfcf23c0] nal_unit_type: 8(PPS), nal_ref_idc: 3
[h264 @ 0xfcf23c0] nal_unit_type: 6(SEI), nal_ref_idc: 0
[h264 @ 0xfcf23c0] nal_unit_type: 5(IDR), nal_ref_idc: 3
[h264 @ 0xfcf23c0] Format yuv420p chosen by get_format().
[h264 @ 0xfcf23c0] Reinit context to 864x480, pix_fmt: yuv420p
[h264 @ 0xfcf23c0] no picture
cur_dts is invalid st:0 (0) [init:0 i_done:0 finish:0] (this is harmless if it occurs once at the start per stream)
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
[h264 @ 0xfcf23c0] nal_unit_type: 9(AUD), nal_ref_idc: 0
[h264 @ 0xfcf23c0] nal_unit_type: 1(Coded slice of a non-IDR picture), nal_ref_idc: 2
[h264 @ 0xfcf23c0] no picture
cur_dts is invalid st:0 (0) [init:0 i_done:0 finish:0] (this is harmless if it occurs once at the start per stream)
[mpegts @ 0xfb9e440] pid=100 pes_code=0x1e0
[h264 @ 0xfcf23c0] nal_unit_type: 9(AUD), nal_ref_idc: 0
[h264 @ 0xfcf23c0] nal_unit_type: 1(Coded slice of a non-IDR picture), nal_ref_idc: 0
detected 1 logical cores
[graph 0 input from stream 0:0 @ 0xfd49dc0] Setting 'video_size' to value '854x480'
[graph 0 input from stream 0:0 @ 0xfd49dc0] Setting 'pix_fmt' to value '0'
[graph 0 input from stream 0:0 @ 0xfd49dc0] Setting 'time_base' to value '1/90000'
[graph 0 input from stream 0:0 @ 0xfd49dc0] Setting 'pixel_aspect' to value '1/1'
[graph 0 input from stream 0:0 @ 0xfd49dc0] Setting 'frame_rate' to value '24/1'
[graph 0 input from stream 0:0 @ 0xfd49dc0] w:854 h:480 pixfmt:yuv420p tb:1/90000 fr:24/1 sar:1/1
[format @ 0xfcdd940] Setting 'pix_fmts' to value 'rgb24|rgba|rgb48be|rgba64be|pal8|gray|ya8|gray16be|ya16be|monob'
[auto_scaler_0 @ 0xfbc2340] Setting 'flags' to value 'bicubic'
[auto_scaler_0 @ 0xfbc2340] w:iw h:ih flags:'bicubic' interl:0
[format @ 0xfcdd940] auto-inserting filter 'auto_scaler_0' between the filter 'Parsed_null_0' and the filter 'format'
[AVFilterGraph @ 0xfe08dc0] query_formats: 5 queried, 3 merged, 1 already done, 0 delayed
[auto_scaler_0 @ 0xfbc2340] picking rgb24 out of 10 ref:yuv420p alpha:0
[auto_scaler_0 @ 0xfbc2340] w:854 h:480 fmt:yuv420p sar:1/1 -> w:854 h:480 fmt:rgb24 sar:1/1 flags:0x4
Output #0, image2, to 'test.png':
  Metadata:
    encoder         : Lavf58.45.100
    Stream #0:0, 0, 1/24: Video: png, 1 reference frame, rgb24(left), 854x480 [SAR 1:1 DAR 427:240], 0/1, q=2-31, 200 kb/s, 24 fps, 24 tbn, 24 tbc
    Metadata:
      encoder         : Lavc58.91.100 png
Clipping frame in rate conversion by 0.000008
[image2 @ 0xfe05b40] Opening 'test.png' for writing
[file @ 0xfcdd880] Setting default whitelist 'file,crypto,data'
[AVIOContext @ 0xfcb63c0] Statistics: 0 seeks, 1 writeouts
No more output streams to write to, finishing.
frame=    1 fps=0.0 q=-0.0 Lsize=N/A time=00:00:00.04 bitrate=N/A speed= 2.5x   
video:19kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: unknown
Input file #0 (offline.ts583003426):
  Input stream #0:0 (video): 3 packets read (4616 bytes); 2 frames decoded;
  Input stream #0:1 (audio): 0 packets read (0 bytes);
  Total: 3 packets (4616 bytes) demuxed
Output file #0 (test.png):
  Output stream #0:0 (video): 1 frames encoded; 1 packets muxed (19734 bytes);
  Total: 1 packets (19734 bytes) muxed
2 frames successfully decoded, 0 decoding errors
[AVIOContext @ 0xfba7340] Statistics: 1331344 bytes read, 2 seeks
```

**Debian 12, failure**:
```
user@debian:~/owncast/data/tmp$ ../../ffmpeg -y -threads 1 -t 1 -i offline.ts122759735 -f image2 -vframes 1 test.png -loglevel trace
ffmpeg version 4.3.1-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2020 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
Splitting the commandline.
Reading option '-y' ... matched as option 'y' (overwrite output files) with argument '1'.
Reading option '-threads' ... matched as AVOption 'threads' with argument '1'.
Reading option '-t' ... matched as option 't' (record or transcode "duration" seconds of audio/video) with argument '1'.
Reading option '-i' ... matched as input url with argument 'offline.ts122759735'.
Reading option '-f' ... matched as option 'f' (force format) with argument 'image2'.
Reading option '-vframes' ... matched as option 'vframes' (set the number of video frames to output) with argument '1'.
Reading option 'test.png' ... matched as output url.
Reading option '-loglevel' ... matched as option 'loglevel' (set logging level) with argument 'trace'.
Finished splitting the commandline.
Parsing a group of options: global .
Applying option y (overwrite output files) with argument 1.
Applying option loglevel (set logging level) with argument trace.
Successfully parsed a group of options.
Parsing a group of options: input url offline.ts122759735.
Applying option t (record or transcode "duration" seconds of audio/video) with argument 1.
Successfully parsed a group of options.
Opening an input file: offline.ts122759735.
[NULL @ 0x6dd3440] Opening 'offline.ts122759735' for reading
[file @ 0x6dd3f00] Setting default whitelist 'file,crypto,data'
Probing mpegts score:50 size:2048
[mpegts @ 0x6dd3440] Format mpegts probed with size=2048 and score=50
[mpegts @ 0x6dd3440] Probe: 8192, score: 44, dvhs_score: 0, fec_score: -2
[mpegts @ 0x6dd3440] Filter: pid=0x11 type=1
[mpegts @ 0x6dd3440] Filter: pid=0x0 type=1
[mpegts @ 0x6dd3440] Filter: pid=0x12 type=1
[mpegts @ 0x6dd3440] SDT:
[mpegts @ 0x6dd3440] tag: 0x48 len=24
Segmentation fault
```

**Comment by gabek** [comment]
Thanks so much for supplying all this detail! It seems at least Owncast isn't to blame here, something specific with ffmpeg is going on where it's no longer able to read a transport stream file. I'm not sure why that would be OS dependent on a statically built binary, but I'll have to do some digging to see what this ffmpeg issue is.

**Comment by glyphscribe** [comment]
I've done a little bit more research on this and it appears that it's not an isolated incident with ffmpeg processing transport stream files, specifically. Linked is a discussion on the _ffmpeg-trac_ bugtracker: https://trac.ffmpeg.org/ticket/10439#comment:2

Excerpt of possible relevance:

> When I take a look at the ffprobe header, it shows that it has been compiled with an old version of gcc : 8.3.0.
> Debian 11 comes with gcc 10.X and Debian 12 with gcc 12.X.
> The problem is probably linked to these outdated libraries.

Unfortunately, though the issue was added to the bugtracker in mid-2023, it hasn't been checked in yet, so this is theoretical more than anything else.

Looking at the snippets above, I can confirm that in all of my testing (and current owncast distribution), the static builds of ffmpeg are indeed built against 8.3.0. If nobody else beats me to it, I'll play around with building an updated ffmpeg to confirm and see if the issue abates.

<blockquote><div><strong><a href="https://trac.ffmpeg.org/ticket/10439#comment:2">#10439 (Segmentation fault of ffprobe with the official static build for ts files with Debian 12)
          – FFmpeg</a></strong></div></blockquote>

**Comment by glyphscribe** [comment]
A functional workaround for debian 12 users for the time being is to use the current distribution of ffmpeg provided by the default package manager. It's built with a newer version of gcc and it is _confirmed_ not to throw segfaults while generating thumbnails.

This likely applies to Ubuntu 24 users as well.

```
user@debian:~$ /usr/bin/ffmpeg -version
ffmpeg version 5.1.6-0+deb12u1 Copyright (c) 2000-2024 the FFmpeg developers
built with gcc 12 (Debian 12.2.0-14)
configuration: --prefix=/usr --extra-version=0+deb12u1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
libavutil      57. 28.100 / 57. 28.100
libavcodec     59. 37.100 / 59. 37.100
libavformat    59. 27.100 / 59. 27.100
libavdevice    59.  7.100 / 59.  7.100
libavfilter     8. 44.100 /  8. 44.100
libswscale      6.  7.100 /  6.  7.100
libswresample   4.  7.100 /  4.  7.100
libpostproc    56.  6.100 / 56.  6.100
```

**Comment by DrFrankenspine** [comment]
Can confirm this works. 

I'm running Debian 12 in a Proxmox KVM and can confirm that using the apt-provided ffmpeg works now. Thumbnails are now working in Owncast and my Logs are clear of the "_Unable to generate thumbnail: Unable to generate thumbnail: exit status 139_" is no longer present.

**Comment by gabek** [comment]
Thanks, all, for looking into this and testing! This makes a lot of sense, and it's good to know we're not alone :)

I guess there are a couple of options here.

- The installer script can detect the OS environment and bail if it's Debian 12 based, with a message telling the user to install ffmpeg via apt.
- The installer script can request sudo privileges and install it itself.
- We find some other static builds of ffmpeg to point to.
- We create our own static builds of ffmpeg to point to.

The best option is to replace the Linux static binaries with another one. Some options seem to be:

- https://ffbinaries.com/downloads
- https://github.com/shaka-project/static-ffmpeg-binaries

I haven't tested them, but it might be a good next step.

<blockquote><div><strong><a href="https://ffbinaries.com/downloads">FFbinaries - Download binaries for ffmpeg, ffprobe, ffserver and ffplay (cross-platform: Windows, Mac, Linux)</a></strong></div><div>Download binaries for ffmpeg, ffprobe, ffserver and ffplay (Windows, Mac, Linux)</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/edaf8a5bf2245849d597b0d6c052b001f1528b300921d5e3d039fd5f219e0797/shaka-project/static-ffmpeg-binaries" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/shaka-project/static-ffmpeg-binaries">GitHub - shaka-project/static-ffmpeg-binaries: Static binaries of FFmpeg, for multiple OS & CPU combinations, built from source in a GitHub Actions workflow.</a></strong></div><div>Static binaries of FFmpeg, for multiple OS & CPU combinations, built from source in a GitHub Actions workflow. - shaka-project/static-ffmpeg-binaries</div></blockquote>

**Comment by gabek** [comment]
I haven't had a chance to test with a replacement static build, but I think that's still probably the next step, assuming they're built with a newer version of gcc.

**Comment by gabek** [comment]
It doesn't seem like any of the options are up-to-date with regards to recent versions of ffmpeg. Are we going to have to build our own ffmpeg static binaries for Linux? That would be awful, I really don't have it in me to maintain that. All the different architectures would be a nightmare.

**Comment by gabek** [comment]
It'll require a bunch of testing before we can point the installer at them, but we'll unfortunately now be building, and hosting, our own amd64 and arm64 Linux binaries of ffmpeg. Huzzah.

I'll open up the repo soon to contributions if there are any tweaks to build settings anyone would like to make, since it's our builds, so we can do whatever we want to them. Anything other than ship with Nvidia support, that is.

**Comment by gabek** [comment]
Here's the repo. It's a WIP. It currently relies on emulation to perform the arm64 build, and it only supports two binaries: amd64 and arm64. We don't have problem with darwin binaries currently, so that wasn't a goal. 32bit x86 builds would be more of a hassle, as would older arm builds. Though if anybody wants/needs these, PRs are welcome.

It is TBD when we can switch over to these builds. They require testing by people like all of you to determine if all the codecs and options are available and as expected.

https://github.com/owncast/ffmpeg-builds

<blockquote><img src="https://opengraph.githubassets.com/56f8fc522fd09517739d6a673287f9c36ceee5684d2a48d64825f4ae43704701/owncast/ffmpeg-builds" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/ffmpeg-builds">GitHub - owncast/ffmpeg-builds</a></strong></div><div>Contribute to owncast/ffmpeg-builds development by creating an account on GitHub.</div></blockquote>

**Comment by gabek** [comment]
Apparently these builds aren't working for people, so we're still stuck right now.

**Comment by glyphscribe** [comment]
I'll put in some time to test and document the static builds in varied environments @gabek . If you want to assign it to me even if for the time being, it'll help keep it top of mind for me.

**Comment by gabek** [comment]
That's probably not necessary. I know they're not working for people.

**Comment by Cast-tor** [comment]
I don't know if this will help or if you already know this:
I went thru the pain of self hosting on a raspberry pi 5. Since I,m running 3 active channels, have been doing a lot of experimenting. One of my channels is running at 4500k/1080p. On my laptop, I'm using obs and nvec encoding. I tried passthru on the pi. Had a segment error. If I used the standard lib on the pi, no problem. If I used the standard 264 lib on obs, i could do passthru and standard lib. Since you are use nvec as your obs source, maybe nvec is not as standard as you think. If you can do a pi re-encode with standard lib264 but not passthru, then ?something different and not quite right with obs nvec? Hope I'm not saying anything foolish.

**Comment by gabek** [comment]
Updated custom builds at: https://github.com/owncast/ffmpeg-builds/releases/tag/20251014144538

<blockquote><img src="https://opengraph.githubassets.com/2fc149656e7f74ad3bd0a9b47963fd0a82be537b2a3ad10f77f868e3677c0e47/owncast/ffmpeg-builds/releases/tag/20251014144538" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="/owncast/ffmpeg-builds/releases/tag/20251014144538">Release 20251014144538 · owncast/ffmpeg-builds</a></strong></div><div>Updated static FFmpeg builds
Full Changelog: 2025101...2025101</div></blockquote>

Installer at: https://raw.githubusercontent.com/owncast/owncast.github.io/refs/heads/gek/new-ffmpeg-static-linux-builds/static/install.sh

**Comment by gabek** [comment]
If you want to test the new ffmpeg binaries (Linux amd64/arm64) with the installer:

`curl -s https://raw.githubusercontent.com/owncast/owncast.github.io/refs/heads/gek/new-ffmpeg-static-linux-builds/static/install.sh | bash`

Just make sure you don't already have `ffmpeg` installed, or it won't install it for you.

**Comment by gabek** [comment]
Additionally, I'm also building `vaapi` compatible, non-static binaries (you can't statically link `vaapi` libs), so the plan is once we test out and deploy these static binaries with the installer, to update the installer to allow people to auto-download the `vaapi` version if their Linux machine is configured for it.

**Comment by gabek** [comment]
The above build is core dumping.
`ERRO[2025-11-19T22:45:47Z] Floating point exception (core dumped)`

```
root@ubuntu-s-1vcpu-512mb-10gb-sfo2-01:~/owncast# ./ffmpeg
ffmpeg version 8.0 Copyright (c) 2000-2025 the FFmpeg developers
  built with gcc 13 (Ubuntu 13.3.0-6ubuntu2~24.04)
  configuration: --enable-nonfree --enable-gpl --enable-openssl --enable-libdav1d --enable-libsvtav1 --enable-libx264 --enable-libx265 --enable-libvpx --enable-libxvid --enable-libvidstab --enable-libaom --enable-libzimg --enable-lv2 --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libtheora --enable-libfdk-aac --enable-libjxl --enable-libwebp --enable-libfreetype --enable-libsrt --enable-libzvbi --enable-libzmq --disable-ffnvcodec --disable-vaapi --disable-debug --disable-shared --enable-pthreads --enable-static --enable-version3 --extra-cflags='-I/app/workspace/include -Wno-int-conversion -fPIC -I/app/workspace/include/lilv-0' --extra-ldexeflags='-static -fPIC' --extra-ldflags=-L/app/workspace/lib --extra-libs='-ldl -lpthread -lm -lz' --pkgconfigdir=/app/workspace/lib/pkgconfig --pkg-config-flags=--static --prefix=/app/workspace --extra-version=
  libavutil      60.  8.100 / 60.  8.100
  libavcodec     62. 11.100 / 62. 11.100
  libavformat    62.  3.100 / 62.  3.100
  libavdevice    62.  1.100 / 62.  1.100
  libavfilter    11.  4.100 / 11.  4.100
  libswscale      9.  1.100 /  9.  1.100
  libswresample   6.  1.100 /  6.  1.100
Universal media converter
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

Use -h to get full help or, even better, run 'man ffmpeg'
root@ubuntu-s-1vcpu-512mb-10gb-sfo2-01:~/owncast# ./owncast
INFO[2025-11-19T22:45:46Z] Owncast v0.2.3-linux-64bit (1d439a6430377fd792595fae265eede55ed79563)
ERRO[2025-11-19T22:45:47Z] Floating point exception (core dumped)
ERRO[2025-11-19T22:45:47Z] transcoding error. look at data/logs/transcoder.log to help debug. your copy of ffmpeg may not support your selected codec of libx264 https://owncast.online/docs/codecs/
```

**Comment by gabek** [comment]
Updated builds at https://github.com/owncast/ffmpeg-builds/releases/tag/20251122180033

Installer at https://raw.githubusercontent.com/owncast/owncast.github.io/refs/heads/gek/new-ffmpeg-static-linux-builds/static/install.sh

<blockquote><img src="https://opengraph.githubassets.com/10d3e082ccd843c274ddc3939ded6ce4040d58055657c2010e98cbc27c69fd05/owncast/ffmpeg-builds/releases/tag/20251122180033" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="/owncast/ffmpeg-builds/releases/tag/20251122180033">Release 20251122180033 · owncast/ffmpeg-builds</a></strong></div><div>Updated static FFmpeg builds
Full Changelog: 2025101...2025112</div></blockquote>
<blockquote><img src="https://opengraph.githubassets.com/a1d5f81cb5b649fa73f9b6566bcc0fa106f0477ab60f0a600ce7eacb54b4c0bb/owncast/owncast.github.io" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/owncast/owncast.github.io/blob/gek/new-ffmpeg-static-linux-builds/static/install.sh">owncast.github.io/static/install.sh at gek/new-ffmpeg-static-linux-builds · owncast/owncast.github.io</a></strong></div><div>Owncast's public facing web site.  Documentation and info. - owncast/owncast.github.io</div></blockquote>


```
root@ubuntu-s-1vcpu-512mb-10gb-sfo2-01:~/owncast# ./ffmpeg
ffmpeg version 8.0.1 Copyright (c) 2000-2025 the FFmpeg developers
  built with gcc 13.2.1 (Alpine 13.2.1_git20231014) 20231014
  configuration: --enable-gpl --enable-openssl --enable-libx264 --enable-libx265 --disable-debug --disable-shared --enable-pthreads --enable-static --enable-version3 --extra-cflags='-I/app/workspace/include -Wno-int-conversion -fPIC' --extra-ldexeflags='-static -fPIC' --extra-ldflags=-L/app/workspace/lib --extra-libs='-ldl -lpthread -lm -lz' --pkgconfigdir=/app/workspace/lib/pkgconfig --pkg-config-flags=--static --prefix=/app/workspace --extra-version=
  libavutil      60.  8.100 / 60.  8.100
  libavcodec     62. 11.100 / 62. 11.100
  libavformat    62.  3.100 / 62.  3.100
  libavdevice    62.  1.100 / 62.  1.100
  libavfilter    11.  4.100 / 11.  4.100
  libswscale      9.  1.100 /  9.  1.100
  libswresample   6.  1.100 /  6.  1.100
Universal media converter
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

Use -h to get full help or, even better, run 'man ffmpeg'
root@ubuntu-s-1vcpu-512mb-10gb-sfo2-01:~/owncast# ./owncast
INFO[2025-11-23T02:36:50Z] Owncast v0.2.3-linux-64bit (1d439a6430377fd792595fae265eede55ed79563)
INFO[2025-11-23T02:36:51Z] Web server is listening on port 8080.
INFO[2025-11-23T02:36:51Z] Configure this server by visiting /admin.
```

**Comment by gabek** [comment]
Has anybody who has experienced issues tested with those builds?

**Comment by gabek** [comment]
Darwin builds are crashing and they're the builds I haven't found a way to test in an automated fashion.

**Comment by gabek** [comment]
Updated release available with fixed Darwin builds.

`curl -s https://raw.githubusercontent.com/owncast/owncast.github.io/refs/heads/gek/new-ffmpeg-static-linux-builds/static/install.sh | bash`

As always, if you're having problems, please test this. Please test it even if you're not having issues.

**Comment by gabek** [comment]
These builds are now the default in production.
